test/cypress/integration/group-paying.spec.js

```diff
@@ -87,6 +95,38 @@ describe('Group Payments', () => {
     cy.giLogout()
   })
 
+  // TODO: move these assertions to unit tests.
+  it('period-related getters should work in the waiting period', () => {
+    cy.giLogin(`user1-${userId}`, { bypassUI: true })
+
+    cy.window().its('sbp').then(sbp => {
+      // The getters we're going to test.
+      const { periodStampGivenDate, periodAfterPeriod, periodBeforePeriod } = sbp('state/vuex/getters')
+      const { distributionDate, distributionPeriodLength } = sbp('state/vuex/getters').groupSettings
+      const onePeriodLengthBefore = addTimeToDate(distributionDate, -distributionPeriodLength)
+      const onePeriodLengthAhead = addTimeToDate(distributionDate, distributionPeriodLength)
+      const oneSecondAhead = addTimeToDate(distributionDate, 1000)
+      const oneSecondBefore = addTimeToDate(distributionDate, -1000)
+      /* eslint-disable no-unused-expressions */
+      expect(periodStampGivenDate(new Date()) === onePeriodLengthBefore).to.be.true
+      expect(periodStampGivenDate(distributionDate) === distributionDate).to.be.true
+      expect(periodStampGivenDate(onePeriodLengthAhead) === onePeriodLengthAhead).to.be.true
+      expect(periodStampGivenDate(onePeriodLengthBefore) === onePeriodLengthBefore).to.be.true
+      expect(periodStampGivenDate(oneSecondAhead) === distributionDate).to.be.true
+      expect(periodStampGivenDate(oneSecondBefore) === onePeriodLengthBefore).to.be.true
+
+      expect(periodAfterPeriod(distributionDate) === onePeriodLengthAhead).to.be.true
+      expect(periodAfterPeriod(onePeriodLengthAhead) === undefined).to.be.true
+      expect(periodAfterPeriod(onePeriodLengthBefore) === distributionDate).to.be.true
+
+      expect(periodBeforePeriod(distributionDate) === onePeriodLengthBefore).to.be.true
+      expect(periodBeforePeriod(onePeriodLengthAhead) === distributionDate).to.be.true
+      expect(periodBeforePeriod(onePeriodLengthBefore) === undefined).to.be.true
+      /* eslint-enable no-unused-expressions */
+    })
+    cy.giLogout()
+  })
```

Comment:
The tests were just logged in as user1, then they logged out, and here it's logging in and logging out again. This is going to increase the test time unnecessarily, so can you move these checks into the previous test?